src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -1165,17 +1172,16 @@ public InputStream saveMultiple(final String... images)
       throws DockerException, IOException, InterruptedException {
 
     final WebTarget resource = resource().path("images").path("get");
-    if (images != null) {
-      for (final String image : images) {
-        resource.queryParam("names", urlEncode(image));
-      }
+    for (final String image : images) {
+      resource.queryParam("names", urlEncode(image));
     }
 
     return request(
         GET,
         InputStream.class,
         resource,
-        resource.request(APPLICATION_JSON_TYPE).header("X-Registry-Auth", authHeader(registryAuth))
+        resource.request(APPLICATION_JSON_TYPE).header("X-Registry-Auth", authHeader(
+            registryAuthSupplier.authFor(images[0])))
```

Comment:
Does this endpoint need auth at all? [Docs don't mention it](https://docs.docker.com/engine/api/v1.28/#operation/ImageGetAll).